### PR TITLE
feat(ingestion/powerbi): DatabricksMultiCloud native query support 

### PR DESCRIPTION
**Pull request by sid-acryl**
1. Support for parsing M-Query expression 
```
"let\n    Source = Value.NativeQuery(DatabricksMultiCloud.Catalogs(\"abc.com\", \"/sql/1.0/warehouses/sdassa\", [Catalog=\"fake_catalog\", Database=null, EnableAutomaticProxyDiscovery=null]){[Name=\"sales_db\",Kind=\"Database\"]}[Data], \"select * from public.sale_history#(lf)where creation_timestamp  >= getDate(-3)\", null, [EnableFolding=true]),\n    #\"NewTable\" = Table.TransformColumn(Source,{{\"creation_timestamp\", type date}})\nin\n    #\"NewTable\""
```
2. Remove ANSI escape character from SQL text before parsing it.
3. Fix M-Query pattern issues

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py [hunk 1]
```diff
@@ -1075,6 +1076,27 @@ def create_urn_using_old_parser(self, query: str, server: str) -> Lineage:
...
+            len(data_access_tokens) >= 13
+        ):  # Explicit catalog name is set in Database argument
+            return tree_function.strip_char(data_access_tokens[9])
+
+        if (
+            len(data_access_tokens) >= 6 and data_access_tokens[4] == "Catalog"
+        ):  # use Catalog name is database
+            return tree_function.strip_char(data_access_tokens[5])
+
+        return None
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:1098]: overall this method seems extremely brittle

Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:1170]: Updated the code

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py [hunk 2]
```diff
@@ -1075,6 +1076,27 @@ def create_urn_using_old_parser(self, query: str, server: str) -> Lineage:
...
+        # In-case of DatabricksMultiCloud_SQL the database-name is the catalog name,
+        # and that name is not available in SQL statement
+        if (
+            data_access_tokens[0]
+            != SupportedDataPlatform.DatabricksMultiCloud_SQL.value.powerbi_data_platform_name
+        ):
+            return None
+
+        if (
+            len(data_access_tokens) >= 13
```
[the remark this answers is not in the document]
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:1161]: Updated the code

File: metadata-ingestion/tests/integration/powerbi/test_m_parser.py [hunk 3]
```diff
@@ -961,3 +962,34 @@ def test_snowflake_double_double_quotes():
         data_platform_tables[0].urn
         == "urn:li:dataset:(urn:li:dataPlatform:snowflake,sl_operations.sale.reports,PROD)"
     )
+
+
+def test_databricks_multicloud():
+    q = M_QUERIES[31]
```
[the remark this answers is not in the document]
Comment (sid-acryl) [reply metadata-ingestion/tests/integration/powerbi/test_m_parser.py]: deleted the duplicate one

**Review by anshbansal — APPROVED** [review]

**Pull request merged**